zntrack/core/base.py

```diff
@@ -48,6 +49,21 @@ def __init__(self, **kwargs):
     def __call__(self, *args, **kwargs):
         """Still here for a depreciation warning for migrating to class based ZnTrack"""
 
+    def __init_subclass__(cls, **kwargs):
+        """Add a dataclass-like init if None is provided"""
+
+        # User provides an __init__
+        if cls.__dict__.get("__init__") is not None:
+            return cls
+
+        # attach an automatically generated __init__ if None is provided
+        zn_option_fiels = []
+        for name, item in cls.__dict__.items():
+            if isinstance(item, params):
+                zn_option_fiels.append(name)
+
+        setattr(cls, "__init__", get_auto_init(fields=zn_option_fiels))
```

Comment:
new init is created

Reply:
Committed the "__signature__" addition to this pull request. ZnTrack parameters are now displayed in the signature. Tested with Jupyter Lab and the code given above.

Reply:
Awesome - the final addition for the next release.